src/plugins/mf_priority.cpp

```diff
@@ -252,19 +267,57 @@ static int validate_cb (flux_plugin_t *p,
     if (fairshare == 0)
         return flux_jobtap_reject_job (p, args, "user fairshare value is 0");
 
-    // make sure user has not already hit their max active jobs count
-    if (max_jobs > 0 && current_jobs >= max_jobs)
-        return flux_jobtap_reject_job (p, args,
-                                       "user has max number of jobs submitted");
-
     if (flux_jobtap_job_aux_set (p,
                                  FLUX_JOBTAP_CURRENT_JOB,
                                  "mf_priority:bank_info",
                                  &bank_it->second,
                                  NULL) < 0)
         flux_log_error (h, "flux_jobtap_job_aux_set");
 
-    bank_it->second.current_jobs++;
+    return 0;
+}
+
+
+static int depend_cb (flux_plugin_t *p,
+                      const char *topic,
+                      flux_plugin_arg_t *args,
+                      void *data)
+{
+    int userid;
+    long int id;
+    struct bank_info *b;
+
+    flux_t *h = flux_jobtap_get_flux (p);
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:i, s:I}",
+                                "userid", &userid, "id", &id) < 0) {
+        flux_log (h,
+                  LOG_ERR,
+                  "flux_plugin_arg_unpack: %s",
+                  flux_plugin_arg_strerror (args));
+        return -1;
+    }
+
+    b = static_cast<bank_info *> (flux_jobtap_job_aux_get (
+                                                    p,
+                                                    FLUX_JOBTAP_CURRENT_JOB,
+                                                    "mf_priority:bank_info"));
+
+    if (b == NULL)
+        flux_jobtap_raise_exception (p, FLUX_JOBTAP_CURRENT_JOB, "plugin",
+                                     3, "mf_priority: bank info is missing");
+
+    // if user has already hit their max running jobs count, add a job
+    // dependency to hold job until an already running job has finished
+    if ((b->max_jobs > 0) && (b->current_jobs == b->max_jobs)) {
+        if (flux_jobtap_dependency_add (p, id, "max-jobs-limit") < 0) {
+            flux_log_error (flux_jobtap_get_flux (p),
+                            "flux_jobtap_dependency_add (%lu)",
+                            id);
+        }
+        b->held_jobs.push_back (id);
```

Comment:
If addition of the dependency fails, then the job will proceed to the next state, therefore the `push_back()` at the very least should be skipped.

I think the right approach would be to raise a job exception and `return -1` instead of just logging an error, since otherwise a job could be run and exceed the max-running-jobs limit.